src/backend/executor/nodeModifyTable.c

```diff
@@ -671,31 +672,9 @@ ExecDelete(ItemPointer tupleid,
 			 tupleid->ip_posid,
 			 segid);
 
-	/*
-	 * get information on the (current) result relation
-	 */
-	if (estate->es_result_partitions && planGen == PLANGEN_OPTIMIZER)
-	{
-		Assert(estate->es_result_partitions->part->parrelid);
-
-#ifdef USE_ASSERT_CHECKING
-		Oid parent = estate->es_result_partitions->part->parrelid;
-#endif
-
-		/* Obtain part for current tuple. */
-		resultRelInfo = slot_get_partition(planSlot, estate);
-		estate->es_result_relation_info = resultRelInfo;
```

Comment:
We should take it back anyway. For DELETE it **safe** to remove that line, **but** the we should support consistency with INSERT and UPDATE (for them its crucial to store the pointer to modifyied relation, at least because of ExecInsertIndexTuples, which uses it).